src/App.tsx

```diff
@@ -1,10 +1,73 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+
+enum ErrorMassage {
+  Load = 'Unable to load todos',
+  Empty = 'Title should not be empty',
+  Add = 'Unable to add a todo',
+  Delete = 'Unable to delete a todo',
+  Update = 'Unable to update a todo',
+}
+
+enum FilterQuery {
+  All = 'All',
+  Active = 'Active',
+  Completed = 'Completed',
+}
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [errorMassage, setErrorMassage] = useState('');
+  const [query, setQuery] = useState(FilterQuery.All);
+
+  const getFilteredTodos = (tod: Todo[], qur: string): Todo[] => {
+    let todosCopy = [...tod];
```

Comment:
can we get rid of let
can we return immediately and pass switch case in callback?